contracts/SwappableWrapToken.sol

```diff
@@ -60,18 +60,19 @@ contract SwappableWrapToken is ERC677Token, ISwappableWrapper, IERC20Wrapper, Ow
      * @param _newToken the new token that will act as the wrapped token
      */
     function swapWrap(address _newToken) public onlyOwner returns (bool){
-        require(AddressUtils.isContract(_newToken), "address is not a contract");
+        require(Address.isContract(_newToken), "address is not a contract");
 
         ERC20 originalToken = wrappedToken;
 
         wrappedToken = ERC20(_newToken);
 
-        require(wrappedToken.balanceOf(msg.sender) >= totalSupply_, "Not enough balance of new token");
+        require(wrappedToken.balanceOf(msg.sender) >= totalSupply(), "Not enough balance of new token");
+        require(wrappedToken.allowance(msg.sender, address(this)) >= totalSupply(), "Wrap Contract has insufficient allowance");
 
-        wrappedToken.transferFrom(msg.sender, this, totalSupply_);
-        originalToken.transfer(msg.sender, totalSupply_);
+        require(wrappedToken.transferFrom(msg.sender, address(this), totalSupply()), "New wrapped transferFrom failed");
+        require(originalToken.transfer(msg.sender, totalSupply()), "Orginal wrapped transfer failed");
 
-        emit WrapSwap(originalToken, wrappedToken, msg.sender);
+        emit WrapSwap(address(originalToken), address(wrappedToken), msg.sender);
 
         return true;
```

Comment:
retaining boolean returns to keep consistency with openzepplin library's implementation of erc20